### ISSUE TITLE
MLMG solver fails for 1D simulation

### ISSUE THREAD

**Issue by roelof-groenewald** [ECP-WarpX/WarpX#3123]
This one I probably should have caught earlier but with some recent MCC tests I've been running, I found that in the MLMG solver fails in the example file `WarpX/Examples/Physics_applications/capacitive_discharge/PICMI_inputs_1d.py`.
Specifically the print out I get (with setting `warpx_self_fields_verbosity=2` for the solver) reads:
```
STEP 1 starts ...
--- INFO    : Setting up collisions for electrons with total collision
              0.031408 0.025024
--- INFO    : Setting up collisions for he_ions with total collision
              0.016152 0.000000
MLMG: # of AMR levels: 1
      # of MG levels on the coarsest AMR level: 7
MLMG: Initial rhs               = 73133.21604
MLMG: Initial residual (resid0) = 73133.21604
MLMG: Bottom solve failed.
MLMG: Iteration   1 Fine resid/bnorm = 0
MLMG: Final Iter. 1 resid, resid/bnorm = 0, 0
MLMG: Timers: Solve = 0.000725806 Iter = 0.000580135 Bottom = 0.00012981


**** WARNINGS ******************************************************************
* GLOBAL warning list  after  [ FIRST STEP ]
*
* --> [!  ] [ElectrostaticSolver] [raised once]
*     Max norm of rho is 0
*     @ Raised by: ALL
*
********************************************************************************

STEP 1 ends. TIME = 1.843657817e-10 DT = 1.843657817e-10
Evolve time = 0.034951157 s; This step = 0.034951157 s; Avg. per step = 0.034951157 s

STEP 2 starts ...
MLMG: # of AMR levels: 1
      # of MG levels on the coarsest AMR level: 7
MLMG: Initial rhs               = 505063.4258
MLMG: Initial residual (resid0) = 505063.4258
MLMG: Bottom solve failed.
MLMG: Iteration   1 Fine resid/bnorm = 0
MLMG: Final Iter. 1 resid, resid/bnorm = 0, 0
MLMG: Timers: Solve = 0.000765782 Iter = 0.000639731 Bottom = 0.000136911
STEP 2 ends. TIME = 3.687315634e-10 DT = 1.843657817e-10
Evolve time = 0.054766081 s; This step = 0.019814924 s; Avg. per step = 0.0273830405 s
```

The reason this wasn't found until now is that I used an older version of this script, with a direct Poisson solver, to do the MCC benchmark calculations (see attached file). However we decided to just use the MLMG solver since in 1D it is fast enough (see [this](https://github.com/ECP-WarpX/WarpX/pull/2710#discussion_r779923883) conversation). But I didn't rerun the benchmark calculations with the MLMG solver.

[PICMI_inputs_1d.txt](https://github.com/ECP-WarpX/WarpX/files/8744811/PICMI_inputs_1d.txt)

**Comment by ax3l** [comment]
cc @WeiqunZhang ?

**Comment by ax3l** [comment]
Weiqun is on PTO, but @dpgrote also has a comment from our meeting on a (faster) direct solve for 1D

**Comment by dpgrote** [comment]
This issue will be resolved by PR #3431 which implements a direct solve in 1D.
